File: serve.js
// Servidor local da Plataforma ACELERA
// Uso: node serve.js
// Acesse: http://localhost:3000
const http = require('http');
const fs = require('fs');
const path = require('path');
const PORT = 3000;
const ROOT = __dirname;

const MIME = {
  '.html': 'text/html; charset=utf-8',
  '.css':  'text/css',
  '.js':   'application/javascript',
  '.png':  'image/png',
  '.jpg':  'image/jpeg',
  '.svg':  'image/svg+xml',
  '.ico':  'image/x-icon',
  '.md':   'text/plain',
};

http.createServer((req, res) => {
  let url = req.url === '/' ? '/index.html' : req.url;
  const filePath = path.join(ROOT, url.split('?')[0]);
  const ext = path.extname(filePath);
  
  fs.readFile(filePath, (err, data) => {
    if (err) {
      res.writeHead(404); res.end('Not found: ' + url); return;
    }
    res.writeHead(200, {
      'Content-Type': MIME[ext] || 'text/plain',
      'Cache-Control': 'no-store',
    });
    res.end(data);
  });
}).listen(PORT, () => {
  console.log('\n  \x1b[1;34m🚀 Plataforma ACELERA\x1b[0m — Servidor local iniciado');
  console.log(`  \x1b[32m➜  Local:\x1b[0m   http://localhost:${PORT}`);
  console.log(`  \x1b[32m➜  Login:\x1b[0m   http://localhost:${PORT}/login.html`);
  console.log('\n  Pressione Ctrl+C para parar.\n');
});

File: serve.py
#!/usr/bin/env python3
"""
Servidor local da Plataforma ACELERA
Uso: python serve.py
Acesse: http://localhost:3000
"""
import http.server
import socketserver
import os

PORT = 3000
DIRECTORY = os.path.dirname(os.path.abspath(__file__))

class Handler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)

    def end_headers(self):
        self.send_header('Cache-Control', 'no-store, no-cache')
        super().end_headers()

    def log_message(self, format, *args):
        print(f"  \033[36m{self.address_string()}\033[0m - {format % args}")

with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print(f"\n  \033[1;34m🚀 Plataforma ACELERA\033[0m — Servidor local iniciado")
    print(f"  \033[32m➜  Local:\033[0m   http://localhost:{PORT}")
    print(f"  \033[32m➜  Login:\033[0m   http://localhost:{PORT}/login.html")
    print(f"\n  Pressione Ctrl+C para parar.\n")
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        print("\n\n  Servidor encerrado.")
